sdk/nodejs/runtime/resource.ts

```diff
@@ -244,10 +245,10 @@ export function readResource(
     opts: ResourceOptions,
     sourcePosition?: SourcePosition,
 ): void {
-    const id: Input<ID> | undefined = opts.id;
-    if (!id) {
+    if (!opts.id) {
         throw new Error("Cannot read resource whose options are lacking an ID value");
     }
+    const id: Promise<Input<ID>> = output(opts.id).promise(true);
 
     const label = `resource:${name}[${t}]#...`;
     log.debug(`Reading resource: id=${Output.isInstance(id) ? "Output<T>" : id}, t=${t}, name=${name}`);
```

Comment:
id is a promise now. It will never be an Output at this point, but also there's not much value in printing a promise object to the debug logs. We could match on opts.id being a value, promise, or output but probably easier to just remove it from the log line here.